### PR TITLE
Add dynamic Windows desktops to terraform provider

### PR DESCRIPTION
**Pull request by probakowski**
Related to https://github.com/gravitational/teleport/issues/37103

changelog: Added dynamic Windows desktops to terraform provider

### CODE REVIEW

File: integrations/terraform/examples/resources/teleport_dynamic_windows_desktop/resource.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Teleport Database
+
+resource "teleport_dynamic_windows_desktop" "example" {
```
Comment (hugoShaka) [inline integrations/terraform/examples/resources/teleport_dynamic_windows_desktop/resource.tf:3]:
```suggestion
resource "teleport_dynamic_windows_desktop" "example" {
```

**Review by hugoShaka — APPROVED** [review]

File: integrations/terraform/examples/resources/teleport_dynamic_windows_desktop/resource.tf (hunk 1 above)
Comment (hugoShaka) [reply integrations/terraform/examples/resources/teleport_dynamic_windows_desktop/resource.tf:3]: Note: you'll need to re-render the docs after this change.

**Comment by hugoShaka** [comment]
Should we backport the PR? To which version?

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: lib/services/presets.go [hunk 2]
```diff
@@ -594,6 +594,7 @@ func NewPresetTerraformProviderRole() types.Role {
 							types.KindClusterMaintenanceConfig,
 							types.KindClusterNetworkingConfig,
 							types.KindDatabase,
+							types.KindDynamicWindowsDesktop,
```
[the remark this answers is not in the document]
Comment (probakowski) [reply lib/services/presets.go:599]: You're right, we need `WindowsDesktopLabels`, I've added it

**Pull request merged**